azurerm/resource_arm_private_link_endpoint.go

```diff
@@ -169,18 +224,86 @@ func resourceArmPrivateLinkEndpointDelete(d *schema.ResourceData, meta interface
 		if response.WasNotFound(future.Response()) {
 			return nil
 		}
-		return fmt.Errorf("Error deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+		return fmt.Errorf("Error deleting Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
 		if !response.WasNotFound(future.Response()) {
-			return fmt.Errorf("Error waiting for deleting Private Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
+			return fmt.Errorf("Error waiting for deleting Private Link Endpoint %q (Resource Group %q): %+v", name, resourceGroup, err)
 		}
 	}
 
 	return nil
 }
 
+func expandArmPrivateLinkEndpointServiceConnection(input []interface{}, parseManual bool) *[]network.PrivateLinkServiceConnection {
+	results := make([]network.PrivateLinkServiceConnection, 0)
+	for _, item := range input {
+		v := item.(map[string]interface{})
+		privateConnectonResourceId := v["private_connection_resource_id"].(string)
+		subresourceNames := v["subresource_names"].([]interface{})
+		requestMessage := v["request_message"].(string)
+		isManual := v["is_manual_connection"].(bool)
+		name := v["name"].(string)
+
+		if isManual == parseManual {
+			result := network.PrivateLinkServiceConnection{
+				Name: utils.String(name),
+				PrivateLinkServiceConnectionProperties: &network.PrivateLinkServiceConnectionProperties{
+					GroupIds:             utils.ExpandStringSlice(subresourceNames),
+					PrivateLinkServiceID: utils.String(privateConnectonResourceId),
+				},
+			}
+
+			if requestMessage != "" {
+				result.PrivateLinkServiceConnectionProperties.RequestMessage = utils.String(requestMessage)
+			}
+
+			results = append(results, result)
+		}
+	}
+
+	return &results
+}
+
+func flattenArmPrivateLinkEndpointServiceConnection(input *[]network.PrivateLinkServiceConnection) []interface{} {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results
+	}
+
+	for _, item := range *input {
+		v := make(map[string]interface{})
+
+		if name := item.Name; name != nil {
+			v["name"] = *name
+		}
+		if props := item.PrivateLinkServiceConnectionProperties; props != nil {
+			if subresourceNames := props.GroupIds; subresourceNames != nil {
+				v["subresource_names"] = utils.FlattenStringSlice(subresourceNames)
+			}
+			if privateConnectionResourceId := props.PrivateLinkServiceID; privateConnectionResourceId != nil {
+				v["private_connection_resource_id"] = *privateConnectionResourceId
+			}
+			if requestMessage := props.RequestMessage; requestMessage != nil {
+				v["request_message"] = *requestMessage
+			}
+			if provisioningState := props.ProvisioningState; provisioningState != "" {
+				v["provisioning_state"] = provisioningState
+			}
+			if s := props.PrivateLinkServiceConnectionState; s != nil {
+				if status := s.Status; status != nil {
+					v["status"] = *status
+				}
+			}
```

Comment:
can we remove this field? Terraform should only be returning when this is fully provisioned

Reply:
This resource doesn't work like a normal resource, if you pass the manual flag it is "created" but it isn't usable yet so I am exposing this so the end users will have an idea as to the state of the endpoint.

* `status` - The current status of the privat link endpoint request, possible values will be `Pending`, `Approved`, `Rejected`, or `Disconnected`.